addons/website_sale_coupon/models/sale_order.py

```diff
@@ -8,6 +8,26 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _try_pending_coupon(self):
+        if not request:
+            return
+
+        pending_coupon = request.session.get('pending_coupon', {})
```

Comment:
as discussed add something here to help people understanding the code..
It is too hard to figure what pending_coupon looks like (might be a sign it is overcomplicated?)

```python
"""
session["pending_coupon"] may looks like:
{
    'promo_code': '124979878824',
    'error': 'xxx',  # result of apply_coupon
    'not_found': 'xxx',  # result of apply_coupon
}
"""
```

Reply:
of course, you had to replace 'xxx' by a real example.. not just copy paste this